extensions/geode-modules/src/distributedTest/java/org/apache/geode/modules/util/ClientServerSessionCacheDUnitTest.java

```diff
@@ -153,6 +127,32 @@ public void preCreatedRegionIsNotCopiedToNewlyStartedServers() {
 
   }
 
+  @Test
+  public void startingAClientWithoutServersFails() {
+    final VM client = VM.getVM(2);
+
+    assertThatThrownBy(() -> client.invoke(this::startClientSessionCache))
+        .hasCauseInstanceOf(NoAvailableServersException.class);
+  }
+
+  @Test
+  public void canPreCreateSessionRegionBeforeStartingClient() {
+    final VM server0 = VM.getVM(0);
+    final VM server1 = VM.getVM(1);
+    final VM client = VM.getVM(2);
+
+    server0.invoke(this::startCacheServer);
+    server1.invoke(this::startCacheServer);
+
+    server0.invoke(this::createSessionRegion);
+    server1.invoke(this::createSessionRegion);
```

Comment:
Are these really happening _before_ the client is started (as the method name states)?

Without some explicit synchronization (maybe via the DUnit blackboard and gates) putting these two `invoke()` lines ahead of the `client.invoke()` line does little to ensure that the session region is created before starting the client.

Reply:
That is a very valid point.  I am looking into it now.